### PR TITLE
Task 5 Edit mode

### PR DESCRIPTION
**Pull request by Laranto-spb**
1. Added edit/delete functionality.
2. Made cardDialog universal for add and edit (+moved margins outside component)
 Some questions in comments =>

### CODE REVIEW

File: src/components/card/card-dialog/CardDialog.js [hunk 1]
```diff
@@ -37,6 +41,19 @@ function CardDialog({ addCard }) {
...
+      updCard({
+        id: cardId,
+        title: cardTitle.trim(),
+        body: cardBody.trim(),
+      });
+    }
+    setOpen((prev) => !prev);
+    clearState();
+  };
+
```
Comment (Laranto-spb) [inline src/components/card/card-dialog/CardDialog.js]: Seems that I've almost repeated saveCard method. How to do it better, cause difference is in 2 methods: updCard() or addCard().

Comment (Laranto-spb) [reply src/components/card/card-dialog/CardDialog.js]: Fixed

File: src/components/main/Main.js [hunk 2]
```diff
@@ -31,25 +31,41 @@ function Main() {
...
   };
 
+  const updateCard = (updatedCard) => {
+    const newCards = cards.map((card) => {
+      if (card.id === updatedCard.id) {
+        return updatedCard;
+      }
+      return card;
+    });
+    setCards([...newCards]);
```
Comment (Artsiom-L) [inline src/components/main/Main.js]: The map() method creates a new array, so why are you recreating it here?
And in your case you're applying an ID comparison to each element of the array, maybe there's a better way to do that?

File: src/components/main/Main.js [hunk 3]
```diff
@@ -31,25 +31,41 @@ function Main() {
...
+  };
+
+  const removeCard = (cardId) => {
+    const filtered = cards.filter((card) => card.id !== cardId);
+    setCards([...filtered]);
```
Comment (Artsiom-L) [inline src/components/main/Main.js]: The filter() method creates a new array, so why are you recreating it here?

File: src/components/main/Main.js (hunk 2 above)
Comment (Laranto-spb) [reply src/components/main/Main.js]: fixed

File: src/components/main/Main.js (hunk 3 above)
Comment (Laranto-spb) [reply src/components/main/Main.js]: fixed

**Review by Artsiom-L — APPROVED** [review]

**Pull request merged**